Lib/profiling/sampling/cli.py

```diff
@@ -332,8 +340,10 @@ def _create_collector(format_type, interval, skip_idle):
         raise ValueError(f"Unknown format: {format_type}")
 
     # Gecko format never skips idle (it needs both GIL and CPU data)
+    # and is the only format that uses opcodes for interval markers
     if format_type == "gecko":
         skip_idle = False
+        return collector_class(interval, skip_idle=skip_idle, opcodes=opcodes)
 
     return collector_class(interval, skip_idle=skip_idle)
```

Comment:
```suggestion
    return collector_class(interval, skip_idle=skip_idle, opcodes=opcodes)
```
Why is it only passed when `format_type == "gecko":`?